Add ensure_pow method

lemunozm opened the pull request:
This PR extends #12967 with the `ensure_pow` function. I think this is also an interesting function to have along with the `ensure_ops` family methods.

`ensure_pow` acts similarly to `checked_pow` but returns an `ArithmeticError` instead.

Related #12754

lemunozm commented on primitives/runtime/src/traits.rs:
```diff
@@ -32,11 +32,11 @@ use impl_trait_for_tuples::impl_for_tuples;
-	EnsureFixedPointNumber, EnsureFrom, EnsureInto, EnsureMul, EnsureMulAssign, EnsureOp,
-	EnsureOpAssign, EnsureSub, EnsureSubAssign, IntegerSquareRoot, One, SaturatedConversion,
-	Saturating, UniqueSaturatedFrom, UniqueSaturatedInto, Zero,
+	checked_pow, ensure_pow, AtLeast32Bit, AtLeast32BitUnsigned, Bounded, CheckedAdd, CheckedDiv,
```
I've reexported `checked_pow`. Please, double-check if you agree with it.

ggwpez replied:
I think we will get rid of these anyway in https://github.com/paritytech/substrate/pull/12509

lemunozm replied:
I've read the issue, could be a problem with our last move of `ArithmeticError` to `sp_arithmetic`?

ggwpez replied:
No idea, I think its stale.

ggwpez reviewed (approved):
Please add some minimal success tests or merge in https://github.com/paritytech/substrate/pull/13043 and use the test structure from there.

lemunozm commented:
I've implemented `ensure_pow` as a function because `checked_pow` is also implemented as a function. But maybe can be interesting to follow the same pattern as other `ensure_ops` and make a trait `EnsurePow` with an `ensure_pow` method. What do you think? I'm not sure why in `num-traits` the `checked_pow` is not implemented as a trait TBH.

ggwpez commented:
> I've implemented `ensure_pow` as a function because `checked_pow` is also implemented as a function. But maybe can be interesting to follow the same pattern as other `ensure_ops` and make a trait `EnsurePow` with an `ensure_pow` method. What do you think? I'm not sure why in `num-traits` the `checked_pow` is not implemented as a trait TBH.

Dont know what is better here. Maybe there can be types that are not `CheckedMul` but can still have an `ensure_pow`?  
I cant imagine but that would not work with a free-standing function otherwise.

ggwpez commented:
bot merge

lemunozm commented:
> (Looks like the CI does not notice when you dismiss reviews 🤔)

Yes, it was weird, the review button switched between you and your partner, instead of selecting both 🤷‍♂️. 

Anyways, thanks for your reviews! 🙌🏻

Polkadot-Forum commented:
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v0-9-37/1736/1
